flink-table-store-connector/src/main/java/org/apache/flink/table/store/connector/source/ContinuousFileSplitEnumerator.java

```diff
@@ -91,6 +92,19 @@ private void addSplit(FileStoreSourceSplit split) {
                 .add(split);
     }
 
+    private void addSplitsBack(Collection<FileStoreSourceSplit> splits) {
+        List<FileStoreSourceSplit> backSplits = new ArrayList<>(splits);
```

Comment:
nit: can use `new LinkedList<>(splits).descendingIterator().forEachRemaining(this::addSplitToHead)`.